### ISSUE TITLE
Printer goes idle mid print

### ISSUE THREAD

**Issue by Highcooley** [repetier/Repetier-Firmware#72]
Sorry, me again :-)

I had this issue twice and another RAMBO / Rostock Max user reported the same. After a couple of hours of printing (2-5 h), the printer stops mid print and goes idle. All heaters stay on and Repetier host considers the job to be done (no errors & idle). 

The way I see it, there could be two explanations. One: Neither the firmware nor the host reports any error if the USB connection gets lost. Both just reconnect and the host quits the print job. Two: Somehow, the firmware starts running through the G-Code without executing it at max speed, until the host sent the last line. Then both, the firmware and the host consider the job as done.

**Comment by nixta78** [comment]
Here I am, two years late to this thread's party! But it actually helped me tremendously, so I wanted to post what I found. I just bought a Boots Industries BI V2.5 printer (a cool delta printer from Kickstarter; now they are out of business basically) and had a similar issue described in this thread. The firmware it has is a custom copy of Repetier 0.91. I haven't spent the time to see if it's the newest flavor of 0.91, but I did do a lot of experimentation and came up with a workaround for the bug described above.

Symptoms: I hit Print and it runs about 27 lines of codes and just sits there waiting. The LCD says "Buf: 3" and the laptop shows no communication from the firmware. So the firmware is just waiting for something to happen, but never does. When the steppers finally timeout, communication resumes.

What I had changed: I had re-calibrated the horizontal offset and the rod length, so I wondered if EEPROM had been screwed up. I restored to factory defaults, and it started printing again!

So, I began changing EEPROM settings one by one until I found what caused it. Turns out it isn't one particular thing, it's likely a geometry calculation that gets out of bounds or something. So by altering the horizontal radius and rod length, it caused the firmware to hang up.

Ideal fix: Firmware update (maybe already fixed in 0.91.7 or whatever).

Workaround: I found that if I set Extruder 2's X and Y offsets to 1 (instead of the zeros I had), and also set the tower offsets for X, Y, and Z to 1 (instead of the default zeros) that got me back printing with my custom settings!

Perhaps this will help somebody in a similar boat. These delta printers can be really frustrating to calibrate, not to mention repairing broken parts and eliminating slop, but they are pretty rewarding when they work well. Good luck everybody!

**Comment by frank16** [comment]
I had the issue for quite a while on a K8200 with several different design electronics boards. It is solved now, this is what I did, I hope it helps someone:
1- my computer was a mini pc with a laptop supply. So, it had no earth connection. Nor did the laptop supply of my printer. Your whole printer + pc setup does need one though, not just to fight electrical noise but also for electrical safety. If a supply fails its mains isolation you would appreciate its current going to mains earth via a wire instead of your body. In electrical design a good grounding system always has a single common mounting point for all the minus and earth wires (called star point). In our case this common point is the electronics board of the printer. From this point the grounds go out via the circuit board terminals, then via the minus wires of my new 12V and 24V supplies to the mains earth (internally via the supplies themselves) and via the USB connector and the shielding of the USB cable to my PC. Looks like a star, hence the name...
If both your PC and Printer have earth terminals on their supplies make sure to plug them into the same mains terminal block!!!

2- My PC had a USB3.0 connection to my printer's USB2.0 chip. BIG NO NO. Should work in theory, but the FTDI chip used in the printer control board doesn't seem to like it much (different protocol negotiations that are only supposedly backwards compatible, I think). I used a nice new shielded 3m long USB2 cable into a USB2 socket in my computer.

3. Windows has a power setting for USB that allows it to drop power after a while if it thinks USB is not used. It will start that up again if the driver has new data to send, but who knows how quickly and if the protocol re-negotiations mess with the rather simple serial connection between FTDI chip and printer controller board. I turned that off in the advanced power settings. Not sure that helps, didn't hurt either...

After this I NEVER had an another mid-print stoppage.

So, in short: Use a printer power supply with earth connection if your computer doesn't have one. Use a shielded USB2 cable of no more than 3 meter. DO NOT USE A USB3 output of your computer. Turn of the USB power setting control in Windows advanced power settings.
